packages/cli/test/integration/publicApi/credentials.test.ts

```diff
@@ -0,0 +1,313 @@
+import * as utils from '../shared/utils';
+import type { CredentialPayload, SaveCredentialFunction } from '../shared/types';
+import type { Role } from '../../../src/databases/entities/Role';
+import type { User } from '../../../src/databases/entities/User';
+import * as testDb from '../shared/testDb';
+import { RESPONSE_ERROR_MESSAGES } from '../../../src/constants';
+
+jest.mock('../../../src/telemetry');
+
+let app: express.Application;
+let testDbName = '';
+let globalOwnerRole: Role;
+let globalMemberRole: Role;
+let saveCredential: SaveCredentialFunction;
+
+beforeAll(async () => {
+	app = await utils.initTestServer({ endpointGroups: ['publicApi'], applyAuth: false });
+	const initResult = await testDb.init();
+	testDbName = initResult.testDbName;
+
+	utils.initConfigFile();
+
+	globalOwnerRole = await testDb.getGlobalOwnerRole();
+	globalMemberRole = await testDb.getGlobalMemberRole();
+	const credentialOwnerRole = await testDb.getCredentialOwnerRole();
+	saveCredential = affixRoleToSaveCredential(credentialOwnerRole);
+
+	utils.initTestLogger();
+	utils.initTestTelemetry();
+});
+
+beforeEach(async () => {
+	await testDb.truncate(['User', 'SharedCredentials', 'Credentials'], testDbName);
+});
+
+afterAll(async () => {
+	await testDb.terminate(testDbName);
+});
+
+test('POST /credentials should create credentials', async () => {
+	let ownerShell = await testDb.createUserShell(globalOwnerRole);
+	ownerShell = await testDb.addApiKey(ownerShell);
+
+	const authOwnerAgent = utils.createAgent(app, {
+		apiPath: 'public',
+		version: 1,
+		auth: true,
+		user: ownerShell,
+	});
+	const payloads = [credentialPayload(), credentialPayload()];
+	// @ts-ignore
+	delete payloads[1].nodesAccess;
+
+	// calls need to be in sequence for sqlite (cannot run multiple transactions in parallel)
+	const responses = [];
+	for (const payload of payloads) {
+		responses.push(await authOwnerAgent.post('/credentials').send(payload));
+	}
+
+	await Promise.all(
+		responses.map(async (response, index) => {
+			expect(response.statusCode).toBe(200);
+
+			const { id, name, type } = response.body;
+
+			expect(name).toBe(payloads[index].name);
+			expect(type).toBe(payloads[index].type);
+
+			const credential = await Db.collections.Credentials!.findOneOrFail(id);
+
+			expect(credential.name).toBe(payloads[index].name);
+			expect(credential.type).toBe(payloads[index].type);
+			expect(credential.data).not.toBe(payloads[index].data);
+
+			const sharedCredential = await Db.collections.SharedCredentials!.findOneOrFail({
+				relations: ['user', 'credentials'],
+				where: { credentials: credential },
+			});
+
+			expect(sharedCredential.user.id).toBe(ownerShell.id);
+			expect(sharedCredential.credentials.name).toBe(payloads[index].name);
+		}),
+	);
+});
+
+test('POST /credentials should fail with invalid inputs', async () => {
+	let ownerShell = await testDb.createUserShell(globalOwnerRole);
+	ownerShell = await testDb.addApiKey(ownerShell);
```

Comment:
Utils.createUser() should replace both lines. It creates a member and automatically assigns an API key